client/my-sites/email/form/mailboxes/components/new-mailbox-list/index.tsx

```diff
@@ -80,7 +81,7 @@ interface MailboxListProps {
 }
 
 const NewMailBoxList = (
-	props: MailboxListProps & { children?: JSX.Element }
+	props: MailboxListProps & { children?: ReactNode }
```

Comment:
Out of curiosity for my learning purposes, why are we making this change?

Reply:
The **ReactNode** type is more flexible as allows you to use a logical expression to conditionally render a component, which resolves to a boolean i.e. the return type. Meanwhile **JSX.Element** as a type does not contain, nor does it resolve to a type.

For some similar conversations you can look [here](https://stackoverflow.com/questions/58123398/when-to-use-jsx-element-vs-reactnode-vs-reactelement).